### ISSUE TITLE
sort_unstable_by has no fallback to equal

### ISSUE THREAD

**Issue by AndriBaal** [mourner/delaunator-rs#27]
I use delaunator in a project of mine and sometimes on specific conditions the delaunator crashes because of the partial_cmp of 2 floats without a fallback. Specifically, this needs to be changed:
```rust
fn sortf(f: &mut [(usize, f64)]) {
    f.sort_unstable_by(|&(_, da), &(_, db)| da.partial_cmp(&db).unwrap());
}
```

to:
```rust
fn sortf(f: &mut [(usize, f64)]) {
    f.sort_unstable_by(|&(_, da), &(_, db)| da.partial_cmp(&db).unwrap_or(core::cmp::Ordering::Equal));
}
```

**Comment by mourner** [comment]
@AndriBaal nice catch! Would you mind making a PR with a fix + test?

**Comment by mourner** [comment]
Looks like this would fix #9 too.

**Comment by mourner** [comment]
Fixed by #28
